tpm/structures.go

```diff
@@ -241,6 +242,57 @@ type pubKey struct {
 	Key             []byte
 }
 
+// A symKey is a TPM representation of a symmetric key.
+type symKey struct {
+	AlgID     uint32
+	EncScheme uint16
+	Key       []byte
+}
+
+// TPMMarshal will marshal the symKey into a byte Buffer as specified the TPM spec.
+func (k *symKey) TPMMarshal(out io.Writer) error {
```

Comment:
Maybe only add a custom type and marshaler for the inner key?
The other fields can be handled normally in tpmutil

Reply:
I felt like adding another type that would just be an alias for []byte was a bit ugly so I decided to let the symKey implement the interface instead.

Reply:
But this ends up doing more work for no reason.
Nothing ugly about an alias type for `[]byte`.

Reply:
Just tried making an alias to []byte satisfy the interface. Ran into a problem with pack and unpack parsing structs differently. For unpack we recursively unpack **pointers** to the fields, whereas with pack it packs the fields directly. Thus when unpacking the interface is satisfied but when packing it is not. I only recently picked up go so I'm not too familiar with the internals, but this does not seem to be the case when having a struct satisfy the interface (both struct and *struct will satisfy and interface). That said, in principle I think having entire tpm structs be self marshaled makes more sense than making "new" primitives selfmarshaling.

Reply:
That all sounds like `tpmutil` doesn't do the right thing when marshalling.
There's no reason why we couldn't handle a field implementing an interface when marsahlling.

Can you leave a TODO (for me) to fix this?
I think it's too far into the weeds to block this PR.